### PR TITLE
cleanup: Switch some deprecated TypeDesc names to modern ones

### PR DESCRIPTION
**Pull request by lgritz**

### CODE REVIEW

File: src/include/OSL/oslconfig.h.in [hunk 1]
```diff
@@ -118,6 +118,7 @@ using OIIO::TypePoint;
 using OIIO::TypeVector;
 using OIIO::TypeNormal;
 using OIIO::TypeMatrix;
+using OIIO::TypeMatrix44;
```
Comment (fpsunflower) [inline src/include/OSL/oslconfig.h.in:121]: Is there a difference between `TypeMatrix` and `TypeMatrix44` ?

Comment (fpsunflower) [reply src/include/OSL/oslconfig.h.in:121]: It _seems_ that both are the same, in which case maybe its worth consolidating on the former and remove the using declaration for the later?

**Review by fpsunflower — COMMENTED** [review]
See minor note about cleaning up `TypeMatrix44`

Otherwise LGTM!

File: src/include/OSL/oslconfig.h.in (hunk 1 above)
Comment (lgritz) [reply src/include/OSL/oslconfig.h.in:121]: They both mean the same type. I suppose I can remove this and just change all the Matrix44 references to Matrix.

Comment (lgritz) [reply src/include/OSL/oslconfig.h.in:121]: updated

**Review by fpsunflower — APPROVED** [review]

**Pull request merged**